spec/fixtures/span_types.json

```diff
@@ -0,0 +1,279 @@
+{
+  "__description": {
+    "<type>": "root element for type identified by '<type>'",
+    "<type>.__description": "description for '<type>' (optional)",
+    "<type>.__used_by": "list of agents that use '<type>' to help document alignment (optional)",
+    "<type>.allow_null_subtype": "true to allow null subtype, false by default if omitted",
+    "<type>.allow_unlisted_subtype": "true to allow unlisted subtypes, false by default if omitted",
+    "<type>.subtypes": "root element for sub-types of type '<type>', if omitted or empty subtype must be null, unless 'allow_unlisted_subtype' is set to true",
+    "<type>.subtypes.<subtype>": "sub-type element for <subtype>",
+    "<type>.subtypes.<subtype>.__description": "description of <subtype> subtype (optional)",
+    "<type>.subtypes.<subtype>.__used_by": "list of agents that use <subtype> to help document alignment (optional)"
+  },
+  "app": {
+    "allow_null_subtype": true,
+    "subtypes": {
+      "inferred": {
+        "__description": "Sampling profiler inferred spans",
+        "__used_by": [
+          "java"
+        ]
+      },
+      "controller": {
+        "__description": "Application controller actions",
+        "__used_by": [
+          "ruby"
+        ]
+      },
+      "graphql": {
+        "__description": "Incoming GraphQL requests",
+        "__used_by": [
+          "ruby"
+        ]
+      },
+      "mailer": {
+        "__description": "Application mailer actions",
+        "__used_by": [
+          "ruby"
+        ]
+      },
+      "resource": {
+        "__description": "Application resource actions",
+        "__used_by": [
+          "ruby"
+        ]
+      }
+    }
+  },
+  "custom": {
+    "__description": "API custom instrumentation",
+    "__used_by": [
+      "java", "ruby"
+    ],
+    "allow_null_subtype": true
+  },
+  "db": {
+    "__description": "database span",
+    "allow_unlisted_subtype": true,
```

Comment:
As the subtypes are dynamically set in the Ruby agent, I added this. I've added the types I encountered in our tests but felt that adding this to indicate the dynamic nature might be a good idea too. _WDYT_, @SylvainJuge ?

Reply:
Ideally we want to have all types & sub-types to be properly listed and documented here, thus I'm not sure that having `allow_unlisted_subtype` = `true` makes sense. What are the values for which the Ruby agent can't have the supported databases explicitly listed here ?

Reply:
It's not so much that we can't list the ones we know about. It was more a signifier that there might be possible _undiscovered_ values out there 👽

But I agree. I'll remove the option.